components/layout_2020/dom_traversal.rs

```diff
@@ -89,14 +89,20 @@ where
         });
 
         let threadsafe_node = node.to_threadsafe();
-        let flags = match threadsafe_node.as_element() {
-            Some(element) if element.is_body_element_of_html_element_root() => {
-                FragmentFlags::IS_BODY_ELEMENT_OF_HTML_ELEMENT_ROOT
-            },
-            Some(element) if element.get_local_name() == &local_name!("br") => {
-                FragmentFlags::IS_BR_ELEMENT
-            },
-            _ => FragmentFlags::empty(),
+        let mut flags = FragmentFlags::empty();
+        if let Some(element) = threadsafe_node.as_element() {
+            if element.is_body_element_of_html_element_root() {
+                flags.insert(FragmentFlags::IS_BODY_ELEMENT_OF_HTML_ELEMENT_ROOT);
+            }
+            match element.get_local_name() {
+                &local_name!("br") => {
+                    flags.insert(FragmentFlags::IS_BR_ELEMENT);
+                },
+                &local_name!("table") | &local_name!("th") | &local_name!("td") => {
+                    flags.insert(FragmentFlags::IS_TABLE_TH_OR_TD_ELEMENT);
+                },
```

Comment:
This (and possibly the br?) should only happen in the HTML namespace.

```js
<!DOCTYPE html>
<style>
*|td { display: table-cell}
</style>
<body>
<script>
var td = document.createElementNS("not-html", "td");
var child = document.createElement("div");
td.appendChild(child);
document.body.appendChild(td);
document.body.append(child.offsetParent.tagName);
</script>
</body>
```

Should say `BODY`, not `td`.

On the other hand, `var td = document.createElementNS("http://www.w3.org/1999/xhtml", "td");` should say `TD`.